Support trailing colon in --goto

It'd be useful if `code --goto file:1:2:` works with a trailing comma.

This'd be useful because many tools add a comma after file[:line[:col]]. For example:

Clang:

```
src/main.rs:1:1: error: unknown type name 'fn'
```

Gcc:

```
src/main.rs:1:1: error: unknown type name ‘fn’
```

Ld:

```
ld:src/main.rs: file format not recognized; treating as linker script
```

Bash:

```
src/main.rs: line 1: syntax error near unexpected token `('
```

Awk:

```
awk: src/main.rs:1: (FILENAME=- FNR=1) fatal: function `main' not defined
```

Make:

```
src/main.rs:1: *** missing separator.  Stop.
```

And so on.

--- a/src/vs/platform/environment/node/argvHelper.ts
+++ b/src/vs/platform/environment/node/argvHelper.ts
@@ -42,7 +42,7 @@ function parseAndValidate(cmdLineArgs: string[], reportWarnings: boolean): Nativ
 
 	const args = parseArgs(cmdLineArgs, OPTIONS, reportWarnings ? errorReporter : undefined);
 	if (args.goto) {
-		args._.forEach(arg => assert(/^(\w:)?[^:]+(:\d*){0,2}$/.test(arg), localize('gotoValidation', "Arguments in `--goto` mode should be in the format of `FILE(:LINE(:CHARACTER))`.")));
+		args._.forEach(arg => assert(/^(\w:)?[^:]+(:\d*){0,2}:?$/.test(arg), localize('gotoValidation', "Arguments in `--goto` mode should be in the format of `FILE(:LINE(:CHARACTER))`.")));
 	}
 
 	return args;

--- a/src/vs/workbench/contrib/search/common/search.ts
+++ b/src/vs/workbench/contrib/search/common/search.ts
@@ -148,8 +148,8 @@ export function getOutOfWorkspaceEditorResources(accessor: ServicesAccessor): UR
 	return resources as URI[];
 }
 
-// Supports patterns of <path><#|:|(><line><#|:|,><col?>
-const LINE_COLON_PATTERN = /\s?[#:\(](?:line )?(\d*)(?:[#:,](\d*))?\)?\s*$/;
+// Supports patterns of <path><#|:|(><line><#|:|,><col?><:?>
+const LINE_COLON_PATTERN = /\s?[#:\(](?:line )?(\d*)(?:[#:,](\d*))?\)?:?\s*$/;
 
 export interface IFilterAndRange {
 	filter: string;

